sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -864,9 +849,15 @@ protected Symbol visitSubqueryExpression(SubqueryExpression node, ExpressionAnal
              *
              * However, we support a single column RowType through the SingleColumnTableType.
              */
-            Field field = fields.get(0);
-            SingleColumnTableType singleColumnTableType = new SingleColumnTableType(field.valueType());
-            return new SelectSymbol(relation, singleColumnTableType);
+            DataType innerType = fields.get(0).valueType();
+            SingleColumnTableType dataType = new SingleColumnTableType(innerType);
+            final SelectSymbol.ResultType resultType;
+            if (context.isArrayExpression(node)) {
```

Comment:
I think the SubqueryExpression should only return an array if it is the immediate child of an ArrayComparisonExpression. Otherwise, the intermediate expression has to take care of returning an array result type.

> In other places we usually just use a bool or some kind of stack variable on the context.

What if you have an array expression inside an expression? you'll reset the flag after the first one.